### PR TITLE
Add .vscode/settings.json to Repo

### PR DESCRIPTION
**Pull request by sdixon194**
Adds a .vscode/settings.json file to the repo to add opinionated VS Code settings. 

## Proposed changes:

Adds a .vscode/settings.json file with an exclude list of intelephense so developers using the jetpack repo with VSCode will be able to properly use intelephense without causing performance problems. 

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

pf4qpu-3P-p2

## Does this pull request change what data or activity we track or use?

No.
## Testing instructions:

Not sure how to test this, but I suppose you could nuke your own VS Code Intelephense settings and checkout this PR and see if things continue to work.

### CODE REVIEW

**Review by zinigor — APPROVED** [review]

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{ 
+	"intelephense.files.exclude": [
+	"**/.git/**",
+	"**/.svn/**",
+	"**/.hg/**",
+	"**/CVS/**",
```
Comment (anomiex) [inline .vscode/settings.json:6]: I suspect these aren't needed. We don't use svn, mercurial, or cvs to check out the monorepo.

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	"**/.history/**",
+	"**/vendor/**/vendor/**",
+	"**/packages/**/vendor/**",
+	"**/projects/**/vendor/**",
+	"**/plugins/**/vendor/**",
+	"**/packages/**/wordpress/**",
+	"**/projects/**/wordpress/**",
+	"**/plugins/**/wordpress/**",
+	"**/.cache/**",
+	"**/jetpack_vendor/**/jetpack_vendor/**",
```
Comment (anomiex) [inline .vscode/settings.json:20]: Seems pointless given line 21?

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+	"**/CVS/**",
+	"**/.DS_Store/**",
+	"**/node_modules/**",
+	"**/bower_components/**",
+	"**/vendor/**/{Tests,tests}/**",
+	"**/.history/**",
+	"**/vendor/**/vendor/**",
+	"**/packages/**/vendor/**",
+	"**/projects/**/vendor/**",
+	"**/plugins/**/vendor/**",
```
Comment (anomiex) [inline .vscode/settings.json:15]: At this point it seems like you might just go ahead and list `"**/vendor/**"`.

That would also make lines 10 and 12 redundant.

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+	"**/packages/**/wordpress/**",
+	"**/projects/**/wordpress/**",
+	"**/plugins/**/wordpress/**",
```
Comment (anomiex) [inline .vscode/settings.json:18]: Same-ish.

File: .vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
+{ 
+	"intelephense.files.exclude": [
+	"**/.git/**",
+	"**/.svn/**",
+	"**/.hg/**",
+	"**/CVS/**",
+	"**/.DS_Store/**",
+	"**/node_modules/**",
+	"**/bower_components/**",
```
Comment (anomiex) [inline .vscode/settings.json:9]: We don't use bower either.

**Review by anomiex — COMMENTED** [review]
Some comments, although I'm not all that familiar with vscode stuff so maybe some of this does make sense.

**Pull request merged**